corehq/motech/openmrs/repeater_helpers.py

```diff
@@ -676,6 +663,21 @@ def get_identifier_source_id(requests, identifier_type):
         raise OpenmrsHtmlUiChanged(
             'Domain "{}": Unexpected OpenMRS login page at "{}".'.format(requests.domain_name, response.url)
         )
+    return session
+
+
+@quickcache(['requests.domain_name', 'requests.base_url', 'identifier_type'])
+def get_identifier_source_id(requests, identifier_type, session):
+    """
+    Returns the ID of the identifier source to be used for generating
+    values for identifiers of the given type.
+    """
+    # The idgen module doesn't offer an API to list identifier sources.
+    # Log in using a requests Session, and scrape /module/idgen/manageIdentifierSources.list
+    # (Urgh.)
+
+    response = requests.get('/ws/rest/v1/patientidentifiertype/{}'.format(identifier_type))
```

Comment:
should this also use the session? may not be directly required, but it may be more consistent in this function?

Reply:
This is a call to the (core) OpenMRS REST API. 

The session is just for interacting with the "idgen" module, which doesn't keep with the standards set by the core API. (It's URLs start with "/module/idgen/".)